New/modified entities for excavation

larswik opened the pull request:
Fixes #386 .

larswik committed:
New/modified entities for excavation

Fixes #386 .

SergejMuhic commented on IFC4x3/Constants/f/FACEEXCAVATION_0OQz8Pf7r36AHB8zu64Byl.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FACEEXCAVATION_0OQz8Pf7r36AHB8zu64Byl" Name="FACEEXCAVATION" UniqueId="186bd219-a47d-4318-a44b-23de0610bf2f">
+	<Documentation>A longitudinal underground excavation either of the full face or parts of the full face.</Documentation>
```
```suggestion
	<Documentation>A longitudinal underground excavation either of the full or parts of the face.</Documentation>
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcExcavation/DocEntity.xml:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcExcavation" UniqueId="16107c0c-a0e0-4f60-ad45-4fd438b56863" BaseDefinition="IfcFeatureElementSubtraction" />
```
Should it be abstract?

larswik replied:
I think so since the subclasses are exhaustive as far as I can understand. I propose to not change.

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcExcavation/Documentation.md:
```diff
@@ -0,0 +1,3 @@
+The resulting void from modification of existing terrain or road structure by excavation or by other means of removing material.
+NOTE Definition from ISO 6707-1: void that results from bulk excavation of material.
```
```suggestion
>NOTE Definition from ISO 6707-1: void that results from bulk excavation of material.
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcExcavation/Documentation.md:
```diff
@@ -0,0 +1,3 @@
+The resulting void from modification of existing terrain or road structure by excavation or by other means of removing material.
+NOTE Definition from ISO 6707-1: void that results from bulk excavation of material.
+NOTE The material excavated and either used as fill or discarded as waste is not modelled as Cut, but may be handled as a different concept (Resource) in the future.
```
```suggestion
>NOTE The material excavated and either used as fill or discarded as waste is not modelled as Cut, but may be handled as a different concept (Resource) in the future.
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcEarthworksCutTypeEnum/DocEnumeration.xml:
```diff
@@ -12,6 +12,9 @@
 		<DocConstant xsi:nil="true" href="BASE_EXCAVATION_3mllVcCAzDm8wNwBHnIpxR" />
 		<DocConstant xsi:nil="true" href="USERDEFINED_3UZXzKKD12RotG6dmTF8v" />
 		<DocConstant xsi:nil="true" href="NOTDEFINED_0hF15nkf79uy7wMtnpxz" />
+		<DocConstant xsi:nil="true" href="CONFINEDOPENEXCAVATION_0Czkm0nIL4BwE58wMhaNl5" />
```
Move above USERDEFINED.

larswik replied:
Done.

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcUndergroundExcavation/DocEntity.xml:
```diff
@@ -0,0 +1,17 @@
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="6ca70e73-e455-44ca-94a5-c03a802cf0bd">
+			<Documentation>Either the _PredefinedType_ attribute is unset, or the inherited attribute _ObjectType_ shall be provided, if the _PredefinedType_ is set to USERDEFINED.</Documentation>
+			<Expression>NOT(EXISTS(PredefinedType)) OR
```
If there is no IfcObjectType we should probably use the HasObjectType rule and make PredefinedType mandatory.

larswik replied:
Done for both IfcEarthworksCut and IfcUndergroundExcavation.

larswik committed:
Update IFC4x3/Constants/f/FACEEXCAVATION_0OQz8Pf7r36AHB8zu64Byl.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcExcavation/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcExcavation/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Fixed according to comments

- WR for both IfcEarthworksCut and IfcUndergroundExcavation changed to HasObjectType
- PredefinedType mandatory
- Moved new enums in IfcEarthworksCutTypeEnum above USERDEFINED
- IfcExcavation already abstract

anborr commented on IFC4x3/Constants/a/ANCHOREDOPENEXCAVATION_2KAOTAKnb3hRuWH67Fcj5T.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ANCHOREDOPENEXCAVATION_2KAOTAKnb3hRuWH67Fcj5T" Name="ANCHOREDOPENEXCAVATION" UniqueId="9429874a-5319-43ad-be20-4461cf9ad15d">
```
I keep reading DOPE but that's probably my problem. ;-)

anborr commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcUndergroundExcavation/DocEntity.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcUndergroundExcavation" UniqueId="8037dfb5-11cd-42b1-b3ad-c794016a5985" EntityFlags="32">
```
BaseDefinition missing?

larswik replied:
Yes, thank you for spotting! Fixing now.

larswik replied:
Added BaseDefinition (=IfcExcavation)

larswik committed:
Specified the BaseDefinition that was missing

SergejMuhic commented:
@anborr comments seem resolved also.
